spec/example_app/app/models/product.rb

```diff
@@ -40,4 +40,16 @@ def valid_slug
       errors.add :name, "must have letters or numbers for the URL"
     end
   end
+
+  class Index < Product
+    self.table_name = "products_index"
+
+    def readonly?
+      true
+    end
+
+    def to_model
+      becomes(Product)
```

Comment:
I've never seen `becomes(...)`; is this a normal Rails method?

Reply:
It is! I learned of it... well, when I was writing this code and found that I needed something like it: https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-becomes

The class here is `Product::Index`, which fed to certain helpers is parameterized as `product_index`. So you end up with things like `polymorphic_path(product)` delegating to `product_index_path` instead of `product_path`. To avoid this, ActiveModel provides this `to_model` interface which is used to ask which model should used for parameterization.

(That's how I remember it. It's been a while...)